chaoscenter/graphql/server/graph/probe.resolvers.go

```diff
@@ -0,0 +1,262 @@
+
+	err := authorization.ValidateRole(ctx, projectID,
+		authorization.MutationRbacRules[authorization.UpdateProbe],
+		model.InvitationAccepted.String())
+	if err != nil {
+		return "", err
+	}
+
+	p := probe.NewProbeRepository(projectID)
+	response, err := p.UpdateProbe(ctx, request)
+	if err != nil {
+		logrus.WithFields(logFields).Error(err)
+		return "", err
+	}
+
+	return response, err
+}
+
+func (r *mutationResolver) DeleteProbe(ctx context.Context, probeName string, projectID string) (bool, error) {
+	logFields := logrus.Fields{
+		"projectId": projectID,
+		"probeName": probeName,
+	}
+
+	logrus.WithFields(logFields).WithField("probeID", probeName).Info("request received to delete a probe")
+
+	err := authorization.ValidateRole(ctx, projectID,
+		authorization.MutationRbacRules[authorization.DeleteProbe],
+		model.InvitationAccepted.String())
+	if err != nil {
+		return false, err
+	}
+
+	p := probe.NewProbeRepository(projectID)
+	response, err := p.DeleteProbe(ctx, probeName)
+	if err != nil {
+		logrus.WithFields(logFields).Error(err)
+		return false, err
+	}
+
+	return response, err
+}
+
+func (r *queryResolver) ListProbes(ctx context.Context, projectID string, probeNames []string, filter *model.ProbeFilterInput) ([]*model.Probe, error) {
+	logFields := logrus.Fields{
+		"projectId":  projectID,
+		"probeNames": probeNames,
+	}
+
+	logrus.WithFields(logFields).Info("request received to get probes")
+	err := authorization.ValidateRole(ctx, projectID,
+		authorization.MutationRbacRules[authorization.ListProbes],
+		model.InvitationAccepted.String())
+	if err != nil {
+		return nil, err
+	}
+
+	p := probe.NewProbeRepository(projectID)
+	response, err := p.ListProbes(ctx, probeNames, filter)
+	if err != nil {
+		logrus.WithFields(logFields).Error(err)
+		return nil, err
+	}
+
+	return response, err
+}
+
+func (r *queryResolver) GetProbe(ctx context.Context, projectID string, probeName string) (*model.Probe, error) {
+	logFields := logrus.Fields{
+		"projectId": projectID,
+		"probeName": probeName,
+	}
+
+	logrus.WithFields(logFields).Info("request received to get probe")
+
+	err := authorization.ValidateRole(ctx, projectID,
+		authorization.MutationRbacRules[authorization.GetProbe],
+		model.InvitationAccepted.String())
+	if err != nil {
+		return nil, err
+	}
+
+	p := probe.NewProbeRepository(projectID)
+	response, err := p.GetProbe(ctx, probeName)
+	if err != nil {
+		logrus.WithFields(logFields).Error(err)
+		return nil, err
+	}
+
+	return response, err
+}
+
+func (r *queryResolver) GetProbeYaml(ctx context.Context, projectID string, request model.GetProbeYAMLRequest) (string, error) {
+	logFields := logrus.Fields{
+		"projectId": projectID,
+		"probeName": request.ProbeName,
+	}
+
+	logrus.WithFields(logFields).Info("request received to get probe YAML")
+
+	err := authorization.ValidateRole(ctx, projectID,
+		authorization.MutationRbacRules[authorization.AddProbe],
```

Comment:
Since this is GetProbeYAML this permission could be mapped to `GetProbe` authorization permission